src/main/java/com/google/sps/servlets/SpecificGuideServlet.java

```diff
@@ -48,4 +48,26 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         response.getWriter().println(gson.toJson(guide.get()));
 
     }
+
+    @Override
+    public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException{
+        long id;
+        try{
+            id = getIdFromPath(request.getPathInfo());
+        }catch(NumberFormatException exception){
+            response.setStatus(400);
+            response.getWriter().print(gson.toJson(new Error("The guide id must be a number", 400)));
+            return;
+        }
+
+        try{
+            helper.delete(id);
+        }catch(RuntimeException exception){
```

Comment:
I'm not sure that Datastore actually throws an exception in this case, so I think you could remove this block.

https://googleapis.dev/java/google-cloud-datastore/latest/com/google/cloud/datastore/Datastore.html#delete-com.google.cloud.datastore.Key...-